bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/FileAudioStream.java

```diff
@@ -70,6 +75,36 @@ private static AudioFormat getAudioFormat(File file) throws AudioException {
         }
     }
 
+    private static AudioFormat parseWavFormat(File file) {
+        try (InputStream inputStream = new FileInputStream(file);
+                BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
+                AudioInputStream audioInputStream = AudioSystem.getAudioInputStream(bufferedInputStream);) {
+            javax.sound.sampled.AudioFormat format = audioInputStream.getFormat();
+            String javaSoundencoding = format.getEncoding().toString();
+            String codecPCMSignedOrUnsigned;
+            if (javaSoundencoding.equals(Encoding.PCM_SIGNED.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_SIGNED;
+            } else if (javaSoundencoding.equals(Encoding.PCM_UNSIGNED.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_UNSIGNED;
+            } else if (javaSoundencoding.equals(Encoding.ULAW.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_ULAW;
+            } else if (javaSoundencoding.equals(Encoding.ALAW.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_ALAW;
+            } else {
+                codecPCMSignedOrUnsigned = null;
+            }
+            Integer bitRate = Math.round(format.getFrameRate() * format.getFrameSize()) * format.getChannels();
+            Long frequency = Float.valueOf(format.getSampleRate()).longValue();
+            return new AudioFormat(AudioFormat.CONTAINER_WAVE, codecPCMSignedOrUnsigned, format.isBigEndian(),
+                    format.getSampleSizeInBits(), bitRate, frequency, format.getChannels());
+
+        } catch (UnsupportedAudioFileException | IOException e) {
+            // assume default format
+            return new AudioFormat(AudioFormat.CONTAINER_WAVE, AudioFormat.CODEC_PCM_SIGNED, false, 16, 705600, 44100L,
```

Comment:
Can we refactor this to be a const?

Reply:
Yes, good idea. Done (in the util class)